src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -418,6 +423,59 @@ public void deleteTemplate(String projectId, String templateName) throws IOExcep
         return instanceTemplates;
     }
 
+    /**
+     * Creates persistent disk snapshot for Compute Engine instance.
+     * This method blocks until the operation completes.
+     *
+     * @param projectId Google cloud project id (e.g. my-project).
+     * @param zone Instance's zone.
+     * @param instanceId Instance's name.
+     *
+     * @throws IOException If an error occured in snapshot creation.
+     * @throws InterruptedException If snapshot creation is interrupted.
+     */
+    public void createSnapshot(String projectId, String zone, String instanceId) throws IOException, InterruptedException {
+        Snapshot snapshot = new Snapshot();
+        try {
+            zone = nameFromSelfLink(zone);
+
+            snapshot.setName(instanceId);
+            Operation op = compute.disks().createSnapshot(projectId, zone, instanceId, snapshot).execute();
```

Comment:
3rd parameter is actually disc ID not instance ID. (Its bit misleading)

Reply:
good catch